File: java/src/main/java/com/hydrapayments/sdk/model/SplitRule.java
package com.hydrapayments.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

public class SplitRule {

    private String id;
    private String transactionId;
    private String total;
    private String currency;
    private List<SplitEntry> splits;
    private String sinkAccountId;
    private String status;
    private String createdAt;

    public SplitRule() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    @JsonProperty("transaction_id")
    public String getTransactionId() { return transactionId; }
    @JsonProperty("transaction_id")
    public void setTransactionId(String transactionId) { this.transactionId = transactionId; }

    public String getTotal() { return total; }
    public void setTotal(String total) { this.total = total; }

    public String getCurrency() { return currency; }
    public void setCurrency(String currency) { this.currency = currency; }

    public List<SplitEntry> getSplits() { return splits; }
    public void setSplits(List<SplitEntry> splits) { this.splits = splits; }

    @JsonProperty("sink_account_id")
    public String getSinkAccountId() { return sinkAccountId; }
    @JsonProperty("sink_account_id")
    public void setSinkAccountId(String sinkAccountId) { this.sinkAccountId = sinkAccountId; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    @JsonProperty("created_at")
    public String getCreatedAt() { return createdAt; }
    @JsonProperty("created_at")
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }
}

File: dotnet/src/HydraPayments.Sdk/Models/SplitRule.cs
using System.Text.Json.Serialization;

namespace HydraPayments.Sdk.Models;

public class SplitRule
{
    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;

    [JsonPropertyName("transaction_id")]
    public string? TransactionId { get; set; }

    [JsonPropertyName("total")]
    public string Total { get; set; } = string.Empty;

    [JsonPropertyName("currency")]
    public string Currency { get; set; } = string.Empty;

    [JsonPropertyName("splits")]
    public List<SplitEntry> Splits { get; set; } = new();

    [JsonPropertyName("sink_account_id")]
    public string? SinkAccountId { get; set; }

    [JsonPropertyName("status")]
    public string Status { get; set; } = string.Empty;

    [JsonPropertyName("created_at")]
    public string CreatedAt { get; set; } = string.Empty;
}
